### ISSUE TITLE
files/ignores/globalIgnores don't restrict file discovery

### ISSUE THREAD

**Issue by swwind** [web-infra-dev/rslint#1164]
## Versions
- `@rslint/core`: reproduces on both 0.6.3 and 0.6.4 (latest at time of writing)
- `eslint-plugin-unicorn`: 68.0.0
- Node: v26.3.0
- OS: Linux (Ubuntu 24.04, kernel 6.17)

## Repro
Minimal repro repo: https://github.com/swwind/rslint-fixture-repro

```
git clone git@github.com:swwind/rslint-fixture-repro.git
cd rslint-fixture-repro
pnpm install
pnpm exec rslint --config rslint.config.js
```

Layout:
```
rslint-fixture-repro/
├── rslint.config.js
├── src/
│   └── clean.ts            # valid TS, contains one `null` literal
└── unrelated-fixtures/
    └── broken.js           # deliberately invalid syntax, unrelated to src/
```

`rslint.config.js`:
```js
import { defineConfig } from '@rslint/core';
import unicorn from 'eslint-plugin-unicorn';

export default defineConfig([
  {
    files: ['src/**/*.ts'],
    ignores: ['unrelated-fixtures/**'],
    plugins: { 'unicorn-js': unicorn },
    rules: {
      'unicorn-js/no-null': 'error',
    },
  },
]);
```

`unrelated-fixtures/broken.js` is a file containing deliberately invalid JS syntax (used elsewhere purely as raw test-fixture data, never meant to be parsed as a program).

## Actual output
```
  TypeScript(TS1109)  — [error] Expression expected.
  ╭─┴──────────( unrelated-fixtures/broken.js:23:6 )─────
  │ 22 │  })();
  │ 23 │  ();
  │ 24 │  Employee(name, basepay);
  ╰────────────────────────────────
```
Exit code 1. No `unicorn-js/no-null` diagnostic for `src/clean.ts` is ever printed, and no summary line ("Found N errors...") appears at all.

## Expected output
The `unicorn-js/no-null` violation in `src/clean.ts` should be reported, exactly as it is when `unrelated-fixtures/` is removed (control case):
```
  unicorn-js/no-null  — [error] Use `undefined` instead of `null`.
  ╭─┴──────────( src/clean.ts:2:10 )─────
  │ 1 │  export function example(): unknown {
  │ 2 │    return null; // should trigger unicorn-js/no-null
  │ 3 │  }
  ╰────────────────────────────────

Found 1 error and 0 warnings (linted 2 files with 1 rule in 807ms using 8 threads)
```

## Impact
Any project with even one intentionally-invalid or non-source `.js`/`.ts` file anywhere under the working directory (test fixtures, generated code, scratch files) will have its *entire* lint run silently produce zero rule diagnostics, with no way to exclude the offending file via config.

**Comment by swwind** [comment]
Verified against latest `main` (`a7170f83`, which includes #1227 / f72841dc), running a local build of `@rslint/core` 0.6.5 against this issue's repro repo unchanged.

**With the original config (entry-level `ignores`):** the run no longer aborts — the `unicorn-js/no-null` diagnostic for `src/clean.ts` and the summary line are now reported alongside the `TS1109` for `unrelated-fixtures/broken.js`:

```
  unicorn-js/no-null  — [error] Use `undefined` instead of `null`.   ( src/clean.ts:2:10 )
  TypeScript(TS1109)  — [error] Expression expected.                 ( unrelated-fixtures/broken.js:23:6 )
Found 2 errors and 0 warnings (linted 3 files with 1 rule ...)
```

The remaining `TS1109` is intentional under the discovery semantics #1227 landed, which align with ESLint flat config: the default extension baseline (`**/*.js` etc.) is always selected, and entry-level `ignores` only scope that entry's rules — they don't remove files from discovery. Only global ignores or `.gitignore` do.

**Both supported exclusion mechanisms now work.** With either `globalIgnores(['unrelated-fixtures/**'])` or a `.gitignore` containing `unrelated-fixtures/` (original config untouched), the output matches the expected result from this issue:

```
  unicorn-js/no-null  — [error] Use `undefined` instead of `null`.   ( src/clean.ts:2:10 )
Found 1 error and 0 warnings (linted 3 files with 1 rule ...)
```

Since `globalIgnores` and `.gitignore` now correctly restrict file discovery (the actionable part of the title), closing this as fixed by #1227. Note the repro has no tsconfig, so this also confirms the no-tsconfig path works — see #1212.
